sklearn/kernel_approximation.py

```diff
@@ -1021,9 +1026,14 @@ def fit(self, X, y=None):
         )
 
         # sqrt of kernel matrix on basis vectors
-        U, S, V = svd(basis_kernel)
-        S = np.maximum(S, 1e-12)
-        self.normalization_ = np.dot(U / np.sqrt(S), V)
+        dtype = _find_matching_floating_dtype(basis_kernel, xp=xp)
+        basis_kernel = xp.asarray(basis_kernel, dtype=dtype)
```

Comment:
This should not be needed. If `basis` is an xp array, then `basis_kernel` should also be one. If not there is a problem. This might be fixed by #29475.